amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseTaker.java

```diff
@@ -530,4 +530,12 @@ private void updateAllLeases(Callable<Long> timeProvider)
     public String getWorkerIdentifier() {
         return workerIdentifier;
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public synchronized Collection<Lease> getAllLeases() {
+        return Collections.unmodifiableCollection(new ArrayList<Lease>(allLeases.values()));
```

Comment:
Just a suggestion, minimum requirement for the KCL is Java 8+. So you could get away with `new ArrayList<>(allLeases.values())`

Reply:
Updated